### PR TITLE
Fix wait for full bucket

### PR DESCRIPTION
**Pull request by amper43**

### CODE REVIEW

**Comment by jonyrock** [comment]
@amper43 may I ask you to add a test for this bugfix

File: analytics/analytics/detectors/pattern_detector.py [hunk 1]
```diff
@@ -65,14 +65,18 @@ def detect(self, dataframe: pd.DataFrame, cache: Optional[models.ModelCache]) ->
...
 
+        res = self.detect(self.bucket.data, cache)
         if len(self.bucket.data) >= self.window_size and cache != None:
-            res = self.detect(self.bucket.data, cache)
             excess_data = len(self.bucket.data) - self.max_window_size
             self.bucket.drop_data(excess_data)
-            return res
         
-        return None
+        return res if res else None
```
Comment (VargBurz) [inline analytics/analytics/detectors/pattern_detector.py]: lets use basic if/else construction, cuz its hard to read..

**Comment by amper43** [comment]
@jonyrock  this is not a bug, just avoiding waiting of full bucket and try run detect with available points instead.

File: analytics/analytics/detectors/pattern_detector.py (hunk 1 above)
Comment (jonyrock) [inline analytics/analytics/detectors/pattern_detector.py]:
```suggestion
        if res:
                return res
        else:
                None
```

Comment (amper43) [reply analytics/analytics/detectors/pattern_detector.py]: @VargBurz  fixed

**Pull request merged**